### PR TITLE
fix(performance): optimize ModuleEnvironmentSandbox.getSubTypeIdentifier

### PR DESCRIPTION
**Pull request by keturn**
Always use the class's fully-qualified Java class name as its type identifier, as ReflectionsSandbox does.

The previous implementation led to calling loadClass on *all* of Component's subclasses every time the serializer asked for this type identifier. We encountered scalability problems with that in #4126.

### How to test

This changes identifiers used by serializers for pretty much everything, which could be a big deal!

Those are used for networking and save games.

So while it could break everything, if some things seem to be working, I don't have ideas of particular edge cases to test for.

Ideas welcome here.

### CODE REVIEW

**Comment by Cervator** [comment]
Can confirm that this seems to resolve the performance issue, nice work @keturn! 🎉 

In host/client + client multiplayer I noticed no lag at all for either client _but_ it still took a full 2 minutes for player 2 to join the first time, so there's still something squirrely going on somewhere.

Player 1 could craft perfectly fine both in-hand and via workbench. Player 2 hit a known issue where crafting a crude weapon in a workbench leaves the output item invisible until picked up. Also the durability overlay on an item appears to have gone buggy and takes up way more space (down from the usual space, roughly a full inventory slot's size total) than it is meant to.

On trying to relog player 2 I hit the other issue where I logged in with the camera stuck at 0,0,0. Still was actually in the right location, could break blocks there and everything, but on the 2nd client itself you're stuck in the ground helplessly.

File: engine/src/main/java/org/terasology/persistence/typeHandling/reflection/SerializationSandbox.java [hunk 1]
```diff
@@ -46,7 +33,9 @@
      * @param <T>      The base type whose subtype needs to be identified.
      * @return The unique identifier for {@code subType}.
      */
-    <T> String getSubTypeIdentifier(Class<? extends T> subType, Class<T> baseType);
+    default <T> String getSubTypeIdentifier(Class<? extends T> subType, Class<T> baseType) {
+        return subType.getName();
```
Comment (keturn) [inline engine/src/main/java/org/terasology/persistence/typeHandling/reflection/SerializationSandbox.java:37]: I think there are no longer any implementations of this that use `baseType` at all.

Should we change the interface and remove that parameter entirely?

If so, is this still a subtypeIdentifier, or is it just its Identifier?

Comment (Cervator) [reply engine/src/main/java/org/terasology/persistence/typeHandling/reflection/SerializationSandbox.java:37]: Probably just keep simplifying, if we need something like this in the future we could put it back on. Might still be good to hear from @eviltak 

Additionally serialization and/or the whole ECS likely would majorly change when reintegrated as libraries anyway, so we don't have to aim for perfection right now

**Comment by Cervator** [comment]
Not sure why GitHub didn't notice me merging this - even shows the related merge commit in the activity list 🤔 

Closing this as merged - tested several additional rounds of multiplayer separately with CoreSampleGameplay, ManualLabor, and JoshariasSurvival and submitted two new issues to cover any leftovers: #4152 and #4154

Many thanks @keturn ! Blocker resolved, but sure more stuff we can fix. I didn't see the spawn at 0,0,0 even once during testing at least :-)

**Comment by Cervator** [comment]
Looks like I was missing the move commit locally for some reason despite having fetched 🤔 Okay trying to merge a second time, maybe then this will auto-close as merged.
